src/test/mina_automation/utils.ml

```diff
@@ -30,3 +30,75 @@ let dedup_and_sort_archive_files files : string list =
 let force_kill process =
   Process.send_signal process Core.Signal.kill ;
   Deferred.map (Process.wait process) ~f:Or_error.return
+
+(** [get_memory_usage_mb pid] retrieves the memory usage in megabytes for the process
+  with the given [pid].
+  
+  This function reads the process memory information from [/proc/pid/statm] and
+  extracts the resident set size (RSS) in pages. It then converts this value to
+  megabytes by multiplying by the page size (4096 bytes) and dividing by 1024^2.
+  
+  @param pid The process ID to query memory usage for
+  @return [Some mb] where [mb] is the memory usage in megabytes if successful,
+      [None] if the process doesn't exist or if there's an error reading
+      the memory information *)
+let get_memory_usage_mb pid =
+  let filename = Printf.sprintf "/proc/%d/statm" pid in
+  try
+    let line = In_channel.read_all filename |> String.strip in
+    match String.split ~on:' ' line with
+    | _ :: resident :: _ ->
+        let resident_pages = Int64.of_string resident in
+        let page_size = Int64.of_int 4096 in
+        let bytes = Int64.( * ) resident_pages page_size in
+        let mb = Int64.(to_float bytes /. 1024.0 /. 1024.0) in
+        Some mb
+    | _ ->
+        None
+  with _ -> None
+
+(** [get_memory_usage_mb_of_user_process process] returns the total memory usage
+  in megabytes for all processes matching the specified process name [process].
+
+  This function executes the 'ps' command to retrieve RSS (Resident Set Size) values
+  for all processes matching the given process name, sums them up, and converts the result
+  from kilobytes to megabytes.
```

Comment:
This is wrong. 
`ps -eo comm,rss --no-headers` returns memory took in KiB, not KB. And the conversion dividing 1024 will get you MiB, not MB.